test/support/test_config.h

```diff
@@ -293,11 +293,14 @@
 #    error "TEST_ONLY_HETERO_POLICIES is passed but device backend is not available"
 #endif
 
-// There are issues with stable_sort for libcpp with iterators with a deleted comma operator, disable those tests
-#if (defined(_LIBCPP_VERSION))
-#   define _PSTL_LIBCPP_NO_COMMA_TESTS_BROKEN 1
+//There are issues with stable_sort for libcpp with iterators with a deleted comma operator, disable no_comma compile
+// only tests. Also, since this adds significant amount of code compilation to the build, lets be more conservative
+// about when we try to test this. Since debug build mode and unnamed lambda support dont change the code in any way
+// which should interact with what we are testing for here, lets disable it for those cases for time / build space.
+#if !defined(_LIBCPP_VERSION) && !defined(_DEBUG) && defined(__SYCL_UNNAMED_LAMBDA__)

```

Comment:
Does `defined(__SYCL_UNNAMED_LAMBDA__)` turn off this case for all non-SYCL tests since it is not defined? Is this too restrictive?

Reply:
good point... I'll fix it.